### PR TITLE
[CWS] Add CI visibility to SecAgent functional tests

### PR DESCRIPTION
**Pull request by modernplumbing**
### What does this PR do?

Modeled after https://github.com/DataDog/datadog-agent/pull/14251

If there's security-agent changes, and therefore the `kitchen_test_security_agent_arm64` and `kitchen_test_security_agent_amazonlinux_x64` jobs run, the individual test output is uploaded to the Datadog CI visibility product upon completion of those kitchen jobs.  The end result looks like this: 

image

Note: We don't automatically upload the output of the manual `kitchen_test_security_agent_x64` jobs, or the stress-test job.

Associated change: Terraform a monitor on failures in the `github.com/DataDog/datadog-agent/pkg/security/tests` suite to alert in `#security-and-compliance-agent-ops` so that we don't have to keep looking at the Gitlab page.  https://github.com/DataDog/terraform-config/pull/15352

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by safchain — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

File: .gitlab/functional_test/security_agent.yml [hunk 1]
```diff
@@ -30,8 +44,10 @@ kitchen_test_security_agent_x64:
   variables:
     KITCHEN_ARCH: x86_64
   before_script:
-    - cd $DD_AGENT_TESTING_DIR
+    - pushd $DD_AGENT_TESTING_DIR
     - tasks/kitchen_setup.sh
+  script:
+    - !reference [ .runtime, script ]
```
Comment (KSerrania) [inline .gitlab/functional_test/security_agent.yml:50]: Nit: is this needed?

This job extends `.kitchen_test_security_agent`, and `.kitchen_test_security_agent` already has:
```
  script:
    - !reference [.runtime, script]
 ```

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb [hunk 2]
```diff
@@ -5,229 +5,12 @@
...
-  recursive true
-end
+include_recipe "::functional-tests"
 
-file "#{system_probe_tests_folder}/color_idx" do
-  content node[:color_idx].to_s
-  mode 644
+if node['platform_family'] != 'windows'
+  include_recipe "::stress-tests"
 end
```
Comment (KSerrania) [inline test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb:16]: If I understand correctly, the run lists for the security-agent tests are:
- `security-agent-test`: `["dd-agent-rhel-workaround", "dd-agent-sles-workaround", "dd-security-agent-check"]`. No specific recipes are specified, so the `default.rb` file is used for each cookbook
- `security-agent-stress`: `["dd-agent-rhel-workaround", "dd-agent-sles-workaround", "dd-security-agent-check::stress-tests"]`

Before that change, `security-agent-test` ran `default.rb` only, and `security-agent-stress` ran `stress-tests.rb` only.

Now:
- `security-agent-test` runs `default.rb`, which includes `functional-tests.rb` (which contains the previous contents of `default.rb`), _and also_ `stress-tests.rb` on Linux, which is a change in behavior I didn't expect (I'd have thought `stress-tests.rb` would only be run for the `security-agent-stress` suite).
- `security-agent-stress` still only runs `stress-tests.rb`.

Was this done on purpose?

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/stress-tests.rb [hunk 3]
```diff
@@ -5,22 +5,14 @@
 # Copyright (C) 2020-present Datadog
 #
 
-if node['platform_family'] != 'windows'
-  wrk_dir = '/tmp/security-agent'
+cookbook_file "#{node['common']['work_dir']}/tests/stresssuite" do
+  source "tests/stresssuite"
+  mode '755'
+end
```
Comment (KSerrania) [inline test/kitchen/site-cookbooks/dd-security-agent-check/recipes/stress-tests.rb:11]: Related to the above, when running the `security-agent-stress` test suite, `default.rb` isn't run, so 
```
directory "#{node['common']['work_dir']}/tests" do
  recursive true
end
```
isn't part of the `run_list`. That may be an issue if the directory isn't created when this `cookbook_file` command is run?

File: .gitlab/functional_test/security_agent.yml (hunk 1 above)
Comment (modernplumbing) [reply .gitlab/functional_test/security_agent.yml:50]: good catch!

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/functional-tests.rb [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+
+# system-probe common
+
+system_probe_tests_folder = '/tmp/system-probe-tests'
+
+directory system_probe_tests_folder do
+  recursive true
+end
+
+file "#{system_probe_tests_folder}/color_idx" do
```
Comment (brycekahle) [inline test/kitchen/site-cookbooks/dd-security-agent-check/recipes/functional-tests.rb:270]: FYI this path is changing in https://github.com/DataDog/datadog-agent/pull/15670 so if that gets merged first, you should update this path.

**Review by brycekahle — APPROVED** [review]

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb (hunk 2 above)
Comment (modernplumbing) [reply test/kitchen/site-cookbooks/dd-security-agent-check/recipes/default.rb]: This was a mistake. I updated the run list and recipes to have default.rb be the place for common code, and to have the run lists include both the default.rb and the respective *-tests.rb.

File: test/kitchen/site-cookbooks/dd-security-agent-check/recipes/stress-tests.rb (hunk 3 above)
Comment (modernplumbing) [reply test/kitchen/site-cookbooks/dd-security-agent-check/recipes/stress-tests.rb]: Commented in the other comment, but updated the run list!

**Review by KSerrania — APPROVED** [review]

**Pull request merged**